### PR TITLE
Added Upload API

### PR DESCRIPTION
**Pull request by listiani13**
Added Upload API

### CODE REVIEW

File: src/controller/uploadController.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// @flow
+import File from '../models/FileModel';
+import formidable from 'formidable';
+import fs from 'fs';
+import {insertFileData} from '../globals/helper';
+import path from 'path';
+
+import type {$Request as Req, $Response as Res} from 'express';
+// QUESTION:how to write types for req in multipart form?
+type ExtReq = Req;
```
Comment (danielsukmana) [inline src/controller/uploadController.js:10]: I think a multipart form data request is just another request; it has the same type.

File: src/controller/uploadController.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    res.status(500).json({
+      status: 'SERVER ERROR',
+      message: e.message,
+    });
+  }
+}
+
+export async function uploadFiles(req: ExtReq, res: Res) {
+  // QUESTION: Is it ok to use multipart form? How to write flow types for this formidable thingy?
+  let form = new formidable.IncomingForm();
```
Comment (danielsukmana) [inline src/controller/uploadController.js:69]: Basically, we don't type modules that we don't create. We can use `flow-typed` when available or just leave it be.

File: src/controller/uploadController.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  // QUESTION: is there any better way to do this?
+  form.parse(req, (err, fields, files) => {
```
Comment (danielsukmana) [inline src/controller/uploadController.js:72]: There's [`denodeify`](https://github.com/matthew-andrews/denodeify) which can turn callback styled functions into promises. But with vanilla `formidable` module, we can't do anything about it.

**Review by danielsukmana — APPROVED** [review]
LGTM! (that's abbreviation for "Looks Good to Me" if you're wondering)

File: src/controller/uploadController.js (hunk 1 above)
Comment (listiani13) [reply src/controller/uploadController.js:10]: so, we can just write something like:
 type ReqBody = {
fileName: string,
... } ?
bc when i console.log multipart form, the type of the result is not an object

**Pull request merged**